alf/algorithms/goal_generator.py

```diff
@@ -30,28 +31,40 @@
 
 
 @gin.configurable
-class RandomCategoricalGoalGenerator(Algorithm):
+class RandomCategoricalGoalGenerator(RLAlgorithm):
     """Random Goal Generation Module
 
     This module generates a random categorical goal for the agent
     in the beginning of every episode.
     """
 
-    def __init__(self, num_of_goals, name="RandomCategoricalGoalGenerator"):
+    def __init__(self,
+                 observation_spec,
```

Comment:
For the ```observation_spec```here, I understand this is due to the change of the base class to ```RLAlgorithm```, but still it feels a bit unnatural to take something unused as input.

Reply:
#335

Reply:
> For the `observation_spec`here, I understand this is due to the change of the base class to `RLAlgorithm`, but still it feels a bit unnatural to take something unused as input.

Check out this issue #335

Reply:
You actually use it at line 66.